src/App.js

```diff
@@ -16,60 +16,92 @@ import apiEndpoints from "./config";
 import { cleanJSON } from "./utils";
 
 const App = () => {
-  const [columns, setColumns] = useState([]);
+  const searchParams = new URLSearchParams(window.location.search);
+
+  const [path, setPath] = useState(searchParams.get("path"));
+  const [columns, setColumns] = useState([
+    { items: [], selected: "published" },
+    ...(path?.split(".").map((element) => ({ items: [], selected: element })) ||
+      []),
+  ]);
```

Comment:
It helps to move complex inline logic to separate functions for better readability and maintainability.

```js
const getInitialColumns = (path) => {
  const defaultColumn = { items: [], selected: "published" };
  if (!path) return [defaultColumn];

  const pathElements = path.split(".");
  const columns = pathElements.map((element) => ({ items: [], selected: element }));

  return [defaultColumn, ...columns];
};
```

And then:
```js
  const searchParams = new URLSearchParams(window.location.search);
  const initialPath = searchParams.get("path");

  const [path, setPath] = useState(initialPath);
  const [columns, setColumns] = useState(getInitialColumns(initialPath));
```